### PR TITLE
eureka-editor: 1.21 -> 1.27b

### PR DESCRIPTION
**Pull request by neonfuz**
###### Motivation for this change
Version bump, several new features and improvements.

There is a new main developer since previous release. There has been improvements to openGL rendering. The project seems to use libGLU now instead of libGL so I swapped in that library.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130468` at db90f156 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>eureka-editor</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '-o root ' doesn't match anything in file 'Makefile'
Near pkgs/applications/misc/eureka-editor/default.nix:34:0:
```
   |
34 |     description = "A map editor for the classic DOOM games, and a few related games such as Heretic and Hexen";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/misc/eureka-editor/default.nix:35:5:
```
   |
35 |     license = licenses.gpl2;
   |     ^
```
</li>
  </ul>
</details>

**Comment by neonfuz** [comment]
fixed the two suggestions above and force pushed

File: pkgs/applications/misc/eureka-editor/default.nix [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
+    url = "mirror://sourceforge/eureka-editor/Eureka/${majorVersion}/eureka-${version}-source.tar.gz";
+    sha256 = "075w7xxsgbgh6dhndc1pfxb2h1s5fhsw28yl1c025gmx9bb4v3bf";
   };
 
-  buildInputs = [ fltk zlib xdg-utils libjpeg xorg.libXinerama libGL ];
+  buildInputs = [ fltk zlib xdg-utils libjpeg xorg.libXinerama libGLU ];
 
   enableParallelBuilding = true;
 
   preBuild = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/eureka-editor/default.nix:17]:
```suggestion
  postPatch = ''
```

File: pkgs/applications/misc/eureka-editor/default.nix [hunk 2]
```diff
@@ -1,25 +1,22 @@
-{ lib, stdenv, fetchzip, fltk, zlib, xdg-utils, xorg, libjpeg, libGL }:
+{ lib, stdenv, fetchzip, fltk, zlib, xdg-utils, xorg, libjpeg, libGLU }:
 
 stdenv.mkDerivation rec {
   pname = "eureka-editor";
-  version = "1.21";
-  shortver = "121";
+  version = "1.27b";
+  majorVersion = "1.27";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/eureka-editor/default.nix:6]:
```suggestion
  version = "${majorVersion}b";
  majorVersion = "1.27";
```
just to make sure not one is stale for some reason.

**Review by c0bw3b — APPROVED** [review]
Result of `nixpkgs-review pr 130468` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>eureka-editor</li>
  </ul>
</details>

Builds and runs

**Pull request merged**